src/Microsoft.DotNet.Build.Tasks/ParseTestCoverageInfo.cs

```diff
@@ -0,0 +1,202 @@
+            XmlDocument coverageXml = new XmlDocument();
+            coverageXml.Load(file);
+
+            string testAssemblyName = file.Replace(".coverage.xml", "");
+
+            Log.LogMessage(MessageImportance.Normal, "Processing file: {0}", testAssemblyName);
+
+            // get only the nodes for visited methods and do a bottom up parsing from that.
+            XmlNodeList methods = coverageXml.SelectNodes("/CoverageSession/Modules/Module/Classes/Class/Methods/Method[not(contains(@visited, 'false'))]");
+
+            Dictionary<string, ModuleInfo> visitedModules = new Dictionary<string, ModuleInfo>();
+
+            foreach (XmlNode methodNode in methods)
+            {
+                // method < methods < class < classes < module
+                XmlNode ModuleNode = methodNode.ParentNode.ParentNode.ParentNode.ParentNode;
+                string moduleName = ModuleNode.SelectSingleNode("ModuleName").InnerText;
+
+                ModuleInfo module = null;
+                if (!visitedModules.TryGetValue(moduleName, out module))
+                {
+                    // Initialize module
+                    module = new ModuleInfo(moduleName);
+
+                    // Get files info
+                    XmlNode filesNode = ModuleNode.SelectSingleNode("Files");
+                    List<string> files = new List<string>(filesNode.ChildNodes.Count);
+                    foreach (XmlNode moduleFile in filesNode.ChildNodes)
+                    {
+                        module.Files.Add(moduleFile.Attributes["fullPath"].Value);
+                    }
+
+                    visitedModules.Add(moduleName, module);
+                }
+
+                string methodSignature = methodNode["Name"].InnerText;
+                // Signature format is "ReturnType Class::method(params)"
+                string[] chunks = methodSignature.Split(new string[] { " ", "::", "(", ")" }, StringSplitOptions.RemoveEmptyEntries);
+                string returnType = chunks[0];
+                string className = chunks[1];
+                string methodName = chunks[2];
+                // not used
+                // string methodParams = chunks[3];
+
+                List<string> methodsCovered = null;
+                if (!module.CoveredMethods.TryGetValue(className, out methodsCovered))
+                {
+                    methodsCovered = new List<string>();
+                    module.CoveredMethods.Add(className, methodsCovered);
+                }
+
+                methodsCovered.Add(methodSignature);
+            }
+            testCoverageInfo.Add(testAssemblyName, visitedModules);
+        }
+
+        /// <summary>
+        /// Save parsed information into a new file
+        /// </summary>
+        public void SaveToFile()
+        {
+            XmlWriterSettings settings = new XmlWriterSettings();
+            settings.Indent = true;
+            string outputFile = Path.Combine(CoverageInfoDirectory,"VisitedMethodsReport.xml");
```

Comment:
This should also be an parameter passed from the task to the target. We don't want to force people to recompile task libraries to change this if there is a problem.